django_q/timeout.py

```diff
@@ -0,0 +1,38 @@
+import signal
+
+from django.utils.translation import gettext_lazy as _
+
+from django_q.conf import logger
+
+from .exceptions import TimeoutException
+
+
+class TimeoutHandler:
+    def __init__(self, timeout: int):
+        self._timeout = timeout
+
+    def raise_timeout_exception(self, signum, frame):
+        raise TimeoutException(
+            f"Task exceeded maximum timeout value ({self._timeout} seconds)"
+        )
+
+    def __enter__(self):
+        # if the timeout is -1, then there is no timeout and the task will always keep running until it's done or manually killed
+        if self._timeout == -1:
+            return
+        try:
+            signal.signal(signal.SIGALRM, self.raise_timeout_exception)
+        except ValueError:  # ValueError is raised for Windows users
+            logger.debug(_("SIGALARM is not available on your platform"))
+
+        signal.alarm(self._timeout)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self._timeout == -1:
+            return
+        """When getting out of the timeout, reset the alarm, so it won't trigger"""
+        try:
+            signal.alarm(0)
+            signal.signal(signal.SIGALRM, signal.SIG_DFL)
+        except ValueError:  # ValueError is raised for Windows users
+            logger.debug(_("SIGALARM is not available on your platform"))
```

Comment:
On windows, lines 24, 28, 35 and 36 produce `AttributeError` instead of `ValueError`.

Reply:
> @19greg96 given that this pull request is merged, I suggest creating a regular issue with more details (e.g. version of Windows and version of CPython) about it at https://github.com/django-q2/django-q2/issues . Otherwise this runs the risk of just falling through the cracks and never be fixed, effectively. Thanks 👍

Thank you for the guidance. I created the issue: https://github.com/django-q2/django-q2/issues/222.

Regaring the comments you have linked, I was not clear in my original comment: I understand that `signal.SIGALRM` is not available under non-Unix systems. The issue is that before the `signal.signal` call raises the `ValueError` for the invalid parameter on windows, trying to access this attribue of the signal module raises an `AttributeError`.